### PR TITLE
Setup vale

### PR DESCRIPTION
**Pull request by programmiri**
# Description

This PR introduces [Vale](https://vale.sh/). The setup is meant to be extended over time in order to establish a more strict style guide for our documentation. At the moment, Vale does basic spell checking. 

- Aiven also uses Vale, e.g. in devportal

## This PR
- adds Vale setup (see `.vale.ini` and `.github/vale`
- add "Google" as style package, which applies some rules from the [Google developer documentation style guide](https://developers.google.com/style). (devportal follows this style guide)
- adds a custom Klaw dictionary with custom vocabulary
- adds README for vale setup
- adds GitHub workflow for vale with documentation
- adds customs scripts to run Vale locally
- extends developer documentation with information about Vale

### CODE REVIEW

File: .github/vale/dicts/klaw.dic [hunk 1]
```diff
@@ -0,0 +1,61 @@
+60
```
Comment (mathieu-anderson) [inline .github/vale/dicts/klaw.dic:1]: 60? :o

Also, did you do this manually? Amazing haha!

File: .github/vale/styles/Google/Latin.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+extends: substitution
+message: "Use '%s' instead of '%s'."
+link: 'https://developers.google.com/style/abbreviations'
+ignorecase: true
+level: error
+nonword: true
+action:
+  name: replace
+swap:
```
Comment (mathieu-anderson) [inline .github/vale/styles/Google/Latin.yml:9]: Haha amazing. Will do this in my writing ^^

File: .github/vale/styles/Google/Periods.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+extends: existence
+message: "Don't use periods with acronyms or initialisms such as '%s'."
```
Comment (mathieu-anderson) [inline .github/vale/styles/Google/Periods.yml:2]: I assume this takes into account and omit all the words defined in the dictionaries (such as API)?

**Review by mathieu-anderson — APPROVED** [review]
This is pretty amazing, learned a lot ^^

File: .github/vale/dicts/klaw.dic (hunk 1 above)
Comment (programmiri) [reply .github/vale/dicts/klaw.dic:1]: Yes :D But it was not too bad, I ran the check, copied the console output, took all words that where marked as error and added the ones where it's a correct word.

File: .github/vale/styles/Google/Periods.yml (hunk 3 above)
Comment (programmiri) [reply .github/vale/styles/Google/Periods.yml:2]: Oh, sorry, I should have added that in the PR description. The files in `/Google` are out generated from the package, so it's nothing we can (or should) edit. 

And yes, our Klaw vocabulary is higher prio then this. 

I think in time we'll establish our own rules more, at that point we may don't even need the google package anymore. I thought it's a good baseline for us.

**Comment by programmiri** [comment]
@mathieu-anderson  I sneaked in two commits, one adding info about `vale sync` in the docs and one renaming the docs files from "README" to be specific about their content, that way they are a bit easier to find when searching for files names.

**Review by mathieu-anderson — APPROVED** [review]

**Pull request merged**